test/automation/src/sql/taskPanel.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Code } from '../code';
+import { QuickAccess } from '../quickaccess';
+
+export class TaskPanel {
+
+	private static readonly taskPanelSelector = 'div.pane-body.task-history';
+
+	constructor(private code: Code, private quickAccess: QuickAccess) {
+	}
+
+	async showTaskPanel(): Promise<void> {
+		try {
+			await this.code.waitForElement(TaskPanel.taskPanelSelector);
+		} catch (e) {
+			await this.quickAccess.runCommand('workbench.action.tasks.toggleTasks');
+			await this.code.waitForElement(TaskPanel.taskPanelSelector);
+		}
+	}
+
+	async waitForTaskComplete(task: string): Promise<void> {
+		await this.code.waitForElement(`${TaskPanel.taskPanelSelector} div.label[title="${task}"]`, undefined, 3000); // wait up to 5 minutes for task to complete
```

Comment:
5 minutes is a long time - we should generally strive for our tests to be much quicker than that. Do you think you actually need 5 minutes for this?

Also you should have this be a parameter with a default value so others don't have to change it if they want to use this.

Reply:
I can change it to 1 minute instead. Usually the package installation is pretty quick but could be slow if there are network issues.

Reply:
Just throwing out an idea - can we log how long a successful package install takes in the test? We could look at that number, and then make a relatively educated guess by setting the timeout to some multiple of the expected case to try to account for the slow case, e.g. 3/4x.

Just a thought 😄

Reply:
Yes, we can actually tell from current logs how long the install takes by looking at the timestamps for when the installation begins and when the successful message is found.